xmtp_mls/src/groups/sync.rs

```diff
@@ -281,6 +337,16 @@ impl MlsGroup {
                 }
             }
             IntentKind::SendMessage => {
+                if !Self::is_valid_epoch(
+                    self.context.inbox_id(),
+                    intent.id,
+                    group_epoch,
+                    message_epoch,
+                    3, // max_past_epochs, TODO: expose from OpenMLS MlsGroup
+                ) {
+                    conn.set_group_intent_to_publish(intent.id)?;
```

Comment:
I just realized there is another case we should think about:

What if the recipient is streaming messages (or receiving push notifs) and is also out of date with the current epoch. In that case the original message will decrypt successfully AND they will eventually receive a duplicate when they next sync. I _think_ the message ID will be the same between these retries, but it's probably worth writing a test for that.